fleetspeak/src/e2etesting/lib/setup_components.go

```diff
@@ -0,0 +1,289 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"github.com/golang/protobuf/proto"
+	ptypes "github.com/golang/protobuf/ptypes"
+	duration "github.com/golang/protobuf/ptypes/duration"
+	daemonservicePb "github.com/google/fleetspeak/fleetspeak/src/client/daemonservice/proto/fleetspeak_daemonservice"
+	clientConfigPb "github.com/google/fleetspeak/fleetspeak/src/client/generic/proto/fleetspeak_client_generic"
+	"github.com/google/fleetspeak/fleetspeak/src/common"
+	spb "github.com/google/fleetspeak/fleetspeak/src/common/proto/fleetspeak"
+	cpb "github.com/google/fleetspeak/fleetspeak/src/config/proto/fleetspeak_config"
+	fcpb "github.com/google/fleetspeak/fleetspeak/src/server/components/proto/fleetspeak_components"
+	grpcServicePb "github.com/google/fleetspeak/fleetspeak/src/server/grpcservice/proto/fleetspeak_grpcservice"
+	servicesPb "github.com/google/fleetspeak/fleetspeak/src/server/proto/fleetspeak_server"
+	"google.golang.org/grpc"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+	"time"
+)
+
+var (
+	mysqlDatabase = flag.String("mysql_database", "", "MySQL database name to use")
+	mysqlUsername = flag.String("mysql_username", "", "MySQL username to use")
+	mysqlPassword = flag.String("mysql_password", "", "MySQL password to use")
+)
+
+type componentCmds struct {
+	serverCmd  *exec.Cmd
+	clientCmd  *exec.Cmd
+	serviceCmd *exec.Cmd
+}
+
+func (cc *componentCmds) killAll() {
+	if cc.serviceCmd != nil {
```

Comment:
Should there be a kill here as well?

Reply:
At this step this service just sends 5 messages and waits several seconds for client responses. It finishes itself, so we don't need to kill this process from here. Probably we will add killing of service process on the next step - when sending messages will be initiated from tests with CreateHunt() function for example, not by the service itself like it happens now